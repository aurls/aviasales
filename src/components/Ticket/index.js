import React from 'react';
import PropTypes from 'prop-types';

import {
  formatPrice,
  getDepartureTime,
  getArrivalTime,
  getDuration,
  getTransfersTitle
} from '../../utils/formatters';
import './ticket.scss';

export const Ticket = ({ ticket }) => {
  const getCarrierUrl = (iataCode) => {
    return `http://pics.avs.io/99/36/${iataCode}.png`;
  };

  return (
    <section key={ticket.id} className="ticket">
      <div className="ticket__price">
        {formatPrice(ticket.price)}&nbsp;₽
      </div>
      <div className="ticket__carrier">
        <img className="ticket__carrier-logo"
          src={getCarrierUrl(ticket.carrier)}
          alt={`${ticket.carrier}-logo`} />
      </div>
      {
        ticket.segments.map((segment) => {
          return (
            <div key={`${ticket.id}${segment.duration}`} className="ticket__segment">
              <div className="ticket__time">
                <h3 className="ticket__segment-subtitle">
                  {segment.origin}&nbsp;&ndash;&nbsp;
                  {segment.destination}
                </h3>
                <p className="ticket__segment-info">
                  {getDepartureTime(segment.date)}&nbsp;&ndash;&nbsp;
                  {getArrivalTime(segment.date, segment.duration)}
                </p>
              </div>
              <div className="ticket__duration">
                <h3 className="ticket__segment-subtitle">
                  В&nbsp;пути
                </h3>
                <p className="ticket__segment-info">
                  {getDuration(segment.duration)}
                </p>
              </div>
              <div className="ticket__transfers">
                <h3 className="ticket__segment-subtitle">
                  {getTransfersTitle(segment.stops.length)}
                </h3>
                <p className="ticket__segment-info">
                  {segment.stops.join(', ')}
                </p>
              </div>
            </div>
          );
        })
      }
    </section>
  );
};

Ticket.propTypes = {
  ticket: PropTypes.object.isRequired
};

export default Ticket;
