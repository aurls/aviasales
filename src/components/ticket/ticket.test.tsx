import React from 'react';
import Enzyme from 'enzyme';
import Ticket from './index';
import mockedTicket from '../../store/tests/mockedTicket';

describe('Ticket component', () => {
  let component: Enzyme.CommonWrapper;

  beforeEach(() => {
    component = Enzyme.shallow(
      <Ticket ticket={mockedTicket} />);
  });

  it('Default state', () => {
    expect(component).toMatchSnapshot();
  });
});
