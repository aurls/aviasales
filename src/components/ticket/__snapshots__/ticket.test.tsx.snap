// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket component Default state 1`] = `
<section
  className="ticket"
  key="1"
>
  <div
    className="ticket__price"
  >
    24 769
     Р
  </div>
  <div
    className="ticket__carrier"
  >
    <img
      alt="EY-logo"
      className="ticket__carrier-logo"
      src="http://pics.avs.io/99/36/EY.png"
    />
  </div>
  <div
    className="ticket__segment"
    key="1797"
  >
    <div
      className="ticket__time"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        MOW
         – 
        HKT
      </h3>
      <p
        className="ticket__segment-info"
      >
        23:24
         – 
        12:41
      </p>
    </div>
    <div
      className="ticket__duration"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        В пути
      </h3>
      <p
        className="ticket__segment-info"
      >
        13ч 17м
      </p>
    </div>
    <div
      className="ticket__transfers"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        2 пересадки
      </h3>
      <p
        className="ticket__segment-info"
      >
        SHA, AUH
      </p>
    </div>
  </div>
  <div
    className="ticket__segment"
    key="1775"
  >
    <div
      className="ticket__time"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        MOW
         – 
        HKT
      </h3>
      <p
        className="ticket__segment-info"
      >
        Invalid Date
         – 
        Invalid Date
      </p>
    </div>
    <div
      className="ticket__duration"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        В пути
      </h3>
      <p
        className="ticket__segment-info"
      >
        12ч 55м
      </p>
    </div>
    <div
      className="ticket__transfers"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        1 пересадка
      </h3>
      <p
        className="ticket__segment-info"
      >
        DXB
      </p>
    </div>
  </div>
</section>
`;
